apps/prairielearn/src/middlewares/autoEnroll.ts

```diff
@@ -0,0 +1,32 @@
+import asyncHandler = require('express-async-handler');
+import { idsEqual } from '../lib/id';
+import { insertCheckedEnrollment } from '../models/enrollment';
+
+export default asyncHandler(async (req, res, next) => {
+  // If the user does not currently have access to the course, but could if
+  // they were enrolled, automatically enroll them. However, we will not
+  // attempt to enroll them if they are an instructor (that is, if they have
+  // a specific role in the course or course instance) or if they are
+  // impersonating another user.
+  if (
+    idsEqual(res.locals.user.user_id, res.locals.authn_user.user_id) &&
+    res.locals.authz_data.authn_course_role === 'None' &&
+    res.locals.authz_data.authn_course_instance_role === 'None' &&
+    res.locals.authz_data.authn_has_student_access &&
+    !res.locals.authz_data.authn_has_student_access_with_enrollment
```

Comment:
These were all switched to `authn_`, which I believe is what we want (it's what was used in the previous location of this code).